### PR TITLE
few bugfixes in BUILDLIN.md

### PR DESCRIPTION
**Pull request by evias**
`BUILDLIN.md` uses curl in several places as a _suggested_ tool to download dependencies at specific URLs.

Added curl to the suggested Prerequisites because of that.

### CODE REVIEW

File: BUILDLIN.md [hunk 1]
```diff
@@ -120,14 +121,26 @@ Rocks
 
 Currently ubuntu 18.04 has gflags in version 2.2.1 and snappy in version 1.1.7 which are OK
```
Comment (gimer) [inline BUILDLIN.md]: have you seen this comment? (gflags compilation should not be required)

File: BUILDLIN.md [hunk 2]
```diff
@@ -120,14 +121,26 @@ Rocks
...
+
 rocks
 ```sh
 git clone https://github.com/nemtech/rocksdb.git rocksdb.git
 cd rocksdb.git
 git checkout v6.2.4-nem
 
 mkdir _build && cd _build
-cmake -DCMAKE_BUILD_TYPE=Release -DWITH_TESTS=OFF -DCMAKE_INSTALL_PREFIX=/usr/local ..
+cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr/local -DWITH_TESTS=OFF -DCMAKE_CXX_FLAGS:STRING="-Wno-error=deprecated-copy -Wno-error=pessimizing-move" ..
```
Comment (gimer) [inline BUILDLIN.md]: this looks suspicious,
 1. why did you need this?
 2. what was compiler version used? what was OS?

Comment (evias) [reply BUILDLIN.md]: Sorry for not addressing, I didn't catch these questions.

Turns out I was using older version of gcc and this got fixed with 0.9.1.1 iirc.

File: BUILDLIN.md [hunk 3]
```diff
@@ -127,7 +128,7 @@ cd rocksdb.git
 git checkout v6.2.4-nem
 
 mkdir _build && cd _build
-cmake -DCMAKE_BUILD_TYPE=Release -DWITH_TESTS=OFF -DCMAKE_INSTALL_PREFIX=/usr/local ..
+cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr/local -DWITH_TESTS=OFF ..
```
Comment (gimer) [inline BUILDLIN.md]: no longer needed?

File: BUILDLIN.md [hunk 4]
```diff
@@ -140,7 +141,7 @@ git clone https://github.com/nemtech/catapult-server.git
 cd catapult-server
 
 mkdir _build && cd _build
-cmake -DBOOST_ROOT=~/boost-build-1.71.0 -DCMAKE_BUILD_TYPE=Release -G Ninja ..
+cmake -DBOOST_ROOT=~/boost-build-1.71.0 -DCMAKE_BUILD_TYPE=Release -DGTEST_ROOT=/usr/local/ -G Ninja ..
```
Comment (gimer) [inline BUILDLIN.md:144]: this is incorrect
